web/src/App.js

```diff
@@ -153,12 +159,18 @@ class App extends Component {
 
     try {
       this.setState({ splitMessage: 'Submitting the transaction.' });
-      await contract.methods.split(beneficiary1Address, beneficiary2Address).send({
+      const tx = await contract.methods.split(beneficiary1Address, beneficiary2Address).send({
         from: accounts[0],
         value: web3.utils.toWei(etherToSplit, 'ether')
+      }).on("transactionHash", txHash => {
+        this.setState({ splitMessage: `Transaction ${txHash} submitted, waiting for the network to mine` });
       });
-      const contractBalance = web3.utils.fromWei(await web3.eth.getBalance(contractAddress));
-      this.setState({ splitMessage: `Splitted successfully.`, contractBalance });
+      if (tx.status) {
+        const contractBalance = web3.utils.fromWei(await web3.eth.getBalance(contractAddress));
+        this.setState({ splitMessage: `Transaction ${tx.transactionHash} got mined successfully.`, contractBalance });
+      } else {
+        this.setState({ splitMessage: `Something went wrong.` });
```

Comment:
Could be worth also mentioning `tx.transactionHash` here.

Reply:
Yes, please see a69c8a41ee5797cfb2e83af44a5353aefa4aa78b